src/ui/ui_review_menu_nbgl.c

```diff
@@ -0,0 +1,332 @@
+            break;
+        case APPROVAL_SIMPLE_TRANSACTION:
+            txInfos.fields[0].item = "Hash";
+            txInfos.fields[0].value = fullHash;
+            txInfos.fields[1].item = "From Address";
+            txInfos.fields[1].value = fromAddress;
+            pairList.nbPairs = 2;
+            break;
+        case APPROVAL_PERMISSION_UPDATE:
+            txInfos.fields[0].item = "Hash";
+            txInfos.fields[0].value = fullHash;
+            txInfos.fields[1].item = "From Address";
+            txInfos.fields[1].value = fromAddress;
+            pairList.nbPairs = 2;
+            txInfos.flowTitle = "Review";
+            txInfos.flowSubtitle = "Permission Update";
+            infoLongPress.text = "Confirm Update";
+            break;
+        case APPROVAL_EXCHANGE_CREATE:
+            txInfos.fields[0].item = "Token 1";
+            txInfos.fields[0].value = fullContract;
+            txInfos.fields[1].item = "Amount 1";
+            txInfos.fields[1].value = (char *) G_io_apdu_buffer;
+            txInfos.fields[2].item = "Token 2";
+            txInfos.fields[2].value = toAddress;
+            txInfos.fields[3].item = "Amount 2";
+            txInfos.fields[3].value = (char *) G_io_apdu_buffer + 100;
+            txInfos.fields[4].item = "From Address";
+            txInfos.fields[4].value = fromAddress;
+            pairList.nbPairs = 5;
+            txInfos.flowTitle = "Review Exchange";
+            infoLongPress.text = "Accept and Create\nExchange";
+            break;
+        case APPROVAL_EXCHANGE_TRANSACTION:
+            txInfos.fields[0].item = "Exchange ID";
+            txInfos.fields[0].value = toAddress;
+            txInfos.fields[1].item = "Token pair";
+            txInfos.fields[1].value = fullContract;
+            txInfos.fields[2].item = "Amount";
+            txInfos.fields[2].value = (char *) G_io_apdu_buffer;
+            txInfos.fields[3].item = "Expected";
+            txInfos.fields[3].value = (char *) G_io_apdu_buffer + 100;
+            txInfos.fields[4].item = "From Address";
+            txInfos.fields[4].value = fromAddress;
+            pairList.nbPairs = 5;
+            break;
+        case APPROVAL_EXCHANGE_WITHDRAW_INJECT:
+            txInfos.fields[0].item = "Action";
+            txInfos.fields[0].value = (char *) G_io_apdu_buffer + 100;
+            txInfos.fields[1].item = "Exchange ID";
+            txInfos.fields[1].value = toAddress;
+            txInfos.fields[2].item = "Token Name";
+            txInfos.fields[2].value = fullContract;
+            txInfos.fields[3].item = "Amount";
+            txInfos.fields[3].value = (char *) G_io_apdu_buffer;
+            txInfos.fields[4].item = "From Address";
+            txInfos.fields[4].value = fromAddress;
+            pairList.nbPairs = 5;
+            break;
+        case APPROVAL_WITNESSVOTE_TRANSACTION:
+            uint8_t i;
+            for (i = 0; i < votes_count; i++) {
```

Comment:
add a check on size of `txInfos.fields` array?

Reply:
Done.